src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -76,6 +76,29 @@ public function iOpenProductPage(ProductInterface $product): void
         $this->sharedStorage->set('product_variant', $productVariant);
     }
 
+    /**
+     * @When I view product :product using slug
+     */
+    public function iViewProductUsingSlug(ProductInterface $product): void
+    {
+        /** @var ProductVariantInterface $productVariant */
+        $productVariant = $product->getVariants()->first();
+
+        $this->client->showByIri('/api/v2/shop/products/slug/'.$product->getSlug());
+
+        $this->sharedStorage->set('product', $product);
+        $this->sharedStorage->set('product_variant', $productVariant);
+    }
+
+    /**
+     * @Then I should be redirected to :product product
+     */
+    public function iShouldBeRedirectedToProduct(ProductInterface $product) {
```

Comment:
We are quite strongly following PSR coding style: https://www.php-fig.org/psr/psr-2/, therefore brackets have to be always on the next line. Also, we are adding return types everywhere
```suggestion
    public function iShouldBeRedirectedToProduct(ProductInterface $product): void
     {
```